lib/chef/resource/systemd_unit.rb

```diff
@@ -0,0 +1,43 @@
+#
+# Author:: Nathan Williams (<nath.e.will@gmail.com>)
+# Copyright:: Copyright 2016, Nathan Williams
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/resource/lwrp_base'
+
+class Chef
+  class Resource
+    class SystemdUnit < Chef::Resource::LWRPBase
+      resource_name :systemd_unit
+      identity_attr :unit_name
+
+      actions :enable, :disable, :mask, :unmask,
+              :start, :stop, :restart, :reload
+
+      default_action :nothing
+
+      attribute :unit_name, kind_of: String, default: lazy { "#{name}.#{type}" }
+      attribute :mode, kind_of: String, default: 'system',
+                       equal_to: %w( system user )
+      attribute :type, kind_of: String, required: true, default: 'service',
```

Comment:
Yep, I think in the generic model this isn't needed anymore, and neither are the child resources, which leads to a much simpler and cleaner model.